pageserver/src/tenant/storage_layer.rs

```diff
@@ -119,16 +138,45 @@ impl LayerAccessStatFullDetails {
             access_kind,
         } = self;
         pageserver_api::models::LayerAccessStatFullDetails {
-            when_millis_since_epoch: when
-                .duration_since(UNIX_EPOCH)
-                .expect("better to die in this unlikely case than report false stats")
-                .as_millis(),
+            when_millis_since_epoch: system_time_to_millis_since_epoch(when),
             task_kind: task_kind.into(), // into static str, powered by strum_macros
             access_kind: *access_kind,
         }
     }
 }
 
+impl LayerResidenceStatus {
+    pub fn evicted() -> Self {
+        LayerResidenceStatus::Evicted {
+            timestamp: SystemTime::now(),
+        }
+    }
+
+    pub fn resident(creating: bool) -> Self {
+        LayerResidenceStatus::Resident {
+            timestamp: SystemTime::now(),
+            creating,
+        }
+    }
```

Comment:
```suggestion
    /// Residence status for a new layer file created during this session.
    pub fn created() -> Self {
        LayerResidenceStatus::Resident {
            timestamp: SystemTime::now(),
            created: true,
        }
    }
    
    /// Residence status for a layer file found already on the disk.
    pub fn existing() -> Self {
        LayerResidenceStatus::Resident { timestamp: SystemTime::now(), created: false }
    }
```

Reply:
Granted, we don't really have a word for the time period between `(load|attach) ---> (pageserver shutdown|detach)` but figured it could be a _tenant session_? :)